api/src/main/scala/com/lightbend/lagom/internal/api/broker/TopicFactory.scala

```diff
@@ -15,3 +20,26 @@ import com.lightbend.lagom.javadsl.api.broker.Topic
 trait TopicFactory {
   def create[Message](topicCall: TopicCall[Message]): Topic[Message]
 }
+
+/**
+ * Provider for a topic factory.
+ *
+ * This layer of indirection is provided so that the ServiceClientImplementor doesn't have to directly depend on a
+ * TopicFactory, it can be optional.
```

Comment:
The problem is how to specify the default implementation.  JSR-330 doesn't offer a way to specify default implementations if nothing else is provided. Guice does have an optional parameter on its `Inject` parameter, but I don't want to make us tied to Guice (a lot of Lagom is using Guice, but it shouldn't be, it should use JSR-330 with Play's module abstraction).

One thing that I want to achieve is to be able to use Lagom with no message broker implementation - since maybe you're not using Lagom's message broker.  So we can have a default implementation that does a conditional lookup of the actual implementation from the injector, and throws an exception if not found, but that's just going to find itself when it looks itself up from the injector.  Hence we need another type.